### PR TITLE
Added types for react-simple-oauth2-login

### PR DESCRIPTION
**Pull request by kostassite**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-simple-oauth2-login/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import * as React from 'react';
+
+export interface OAuth2LoginProps {
+    id?: string;
+    authorizationUrl: string;
+    clientId: string;
+    redirectUri: string;
+    responseType: 'code' | 'token';
+    onSuccess?: (data: Record<string, any>) => void;
+    onFailure?: (err: Error) => void;
```
Comment (peterblazejewicz) [inline types/react-simple-oauth2-login/index.d.ts:15]: aren't those required?
https://github.com/bhubr/react-simple-oauth2-login/blob/master/src/OAuth2Login.jsx#L141-L142
```suggestion
    onSuccess: (data: Record<string, any>) => void;
    onFailure: (err: Error) => void;
```

File: types/react-simple-oauth2-login/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        onClick: () => void
+    }) => void;
+    isCrossOrigin?: boolean;
+    onRequest?: () => void;
+    scope?: string;
+    state?: string;
+    extraParams?: Record<string, any>;
+}
+
+export default function OAuth2Login(props: OAuth2LoginProps): JSX.Element;
```
Comment (jakebailey) [inline types/react-simple-oauth2-login/index.d.ts:34]: The upstream package is CJS, so we shouldn't use `export default` here. Additionally, the upstream repo is a class.

I think you should write this like:

```ts
export = OAuth2Login;

declare class OAuth2Login extends React.Component<OAuth2Login.OAuth2LoginProps> {}

declare namespace OAuth2Login {
    interface OAuth2LoginProps {
        // ...
    }
}
```

If you want an example, you check out `react-popover`.

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/react-simple-oauth2-login/index.d.ts (hunk 2 above)
Comment (jakebailey) [reply types/react-simple-oauth2-login/index.d.ts:34]: Apologies, I misread the package; it really does export `default`.

**Review by jakebailey — APPROVED** [review]

**Comment by jakebailey** [comment]
That being said, it does seem like it's a class, not a function, but that's sorta whatever.

**Comment by kostassite** [comment]
Ready to merge

**Pull request merged**